test/uk/gov/hmrc/cipphonenumberverification/services/PasscodeServiceSpec.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cipphonenumberverification.services
+
+import org.mockito.ArgumentMatchersSugar.any
+import org.mockito.IdiomaticMockito
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.libs.json.JsObject
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+import uk.gov.hmrc.cipphonenumberverification.models.{Passcode, PhoneNumber}
+import uk.gov.hmrc.cipphonenumberverification.repositories.PasscodeCacheRepository
+import uk.gov.hmrc.mongo.cache.{CacheItem, DataKey}
+
+import java.time.Instant
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class PasscodeServiceSpec extends AnyWordSpec
+  with Matchers
+  with IdiomaticMockito {
+
+  "persistPasscode" should {
+    //TODO: Since we do not know what the value of the OTP is going to be, we cannot assert the value of the result
```

Comment:
we could know the result if we mock the call to the otp generator class.  you could argue that the otp generator should be mocked anyway.  as the unit test for class x should only test class x.
Then you can assert the passcode etc.

Reply:
Yes, I was thinking the same, we should inject the otp generator into the passcode service. Or, I suppose we could use argument captors